File: Implementierung/frontend/paschi/src/model/userdata/rooms/Chair.ts
import { RoomObject } from "@/model/userdata/rooms/RoomObject";
import { User } from "@/model/User";
import { Position } from "@/model/userdata/rooms/Position";
import { Dimensions } from "@/model/userdata/rooms/Dimensions";

export class Chair extends RoomObject {

  constructor(id: string | undefined, localId: number, user: User, position: Position) {
    super(id, localId, user, position, new Dimensions(500, 500));
  }

  isTable(): boolean {
    return false;
  }
}

File: Implementierung/Backend/src/main/java/edu/kit/informatik/model/userdata/rooms/Chair.java
package edu.kit.informatik.model.userdata.rooms;

import edu.kit.informatik.model.User;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

/**
 * Entität eines Stuhls
 *
 * @author uekai
 * @author ugqbo
 * @version 1.0
 */
@NoArgsConstructor
@Entity
@Table(name = "chairs")
public class Chair extends RoomObject {
    /**
     * Konstruktor zum Erstellen eines Objektes
     * @param user {@link User}
     * @param position {@link Position}
     * @param createdAt {@link Timestamp} der Erstellung
     * @param updatedAt {@link Timestamp} der letzten Änderung
     */
    public Chair(User user, Position position, Timestamp createdAt, Timestamp updatedAt) {
        super(user, position, createdAt, updatedAt);
    }

    @Override
    public boolean isTable() {
        return false;
    }
}
